### ISSUE TITLE
`$stealth` modifier doesn't work correctly in iframes

### ISSUE THREAD

**Issue by AdamWr** [AdguardTeam/AdguardBrowserExtension#2004]
### Issue Details

- **AdGuard version:**
  -  AdGuard  v4.0.133
- **Browser and version:**
  -  Chrome 99.0.4844.74

### Steps to reproduce
1. Enable Stealth mode and `Hide Referrer from third parties` option
2. Add this rule to user rules
```
@@||i.ytimg.com^$stealth,domain=youtube.com
```
3. Go to - https://example.org/
4. Press `F12`, go to `Console` pane and run this script to load iframe with YouTube video:
```js
 (() => {
        let video = '<iframe width="560" height="315" src="https://www.youtube.com/embed/Fy2rtb95QhY" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>';
        let createDiv = document.createElement("div");
        document.body.appendChild(createDiv);
        createDiv.innerHTML = video;
      })();
```
5. Go to network pane and search for:
```
i.ytimg.com
```
6. Click on a search result and in `Headers` tab check `referer`

### Expected Behavior
`@@||i.ytimg.com^$stealth,domain=youtube.com` should disable `stealth` mode and referer should shows:
```
referer: https://www.youtube.com/
```

<details><summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/29142494/158887262-2d4e8db4-78bc-4670-9f52-a19dbebfaaae.png)

</details>

### Actual Behavior
It looks like that `$stealth` rule is not applied to iframe and referer shows:
```
referer: https://i.ytimg.com/
```

<details><summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/29142494/158886899-3cb89384-894a-4802-a417-2e93b278301e.png)

![image](https://user-images.githubusercontent.com/29142494/158887540-8c3ff495-b190-4c78-920f-ff03faf58dd2.png)

</details>

### Additional Information
This rule:
```
@@||i.ytimg.com^$stealth,domain=example.org
```
causes that `stealth` mode is disabled in iframe (in this case `youtube.com`), if I'm not wrong, it's not a correct behavior.

**Comment by stanislav-atr** [comment]
As i understand, `$domain` modifier only matches by main frame, therefore expected behaviour is not true.
@maximtop could you please confirm this?

**Comment by maximtop** [comment]
@stanislav-atr compare please how it behaves in the extension v3.x and in the desktop application

**Comment by maximtop** [comment]
Seems like this was already resolved in the v4.2
